### ISSUE TITLE
[Bug]: [JSON]  Search failed with unexpected error "EMPTY: no JSON found" using expression "json_field[\"number\"] >= 0" if there is list in JSON field

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24118]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230515-30415e1b
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):   all 
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.4.0.dev27
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

 Search failed with unexpected error "EMPTY: no JSON found" using expression "json_field[\"number\"] >= 0"
```
RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=13, reason=worker(13) query failed: [UnexpectedError] Assert "err == simdjson::SUCCESS" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Json.h:104
 => failed to parse the json : EMPTY: no JSON found)>, <Time:{'RPC start': '2023-05-15 18:48:20.331421', 'RPC error': '2023-05-15 18:48:20.465013'}>
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 666, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 521, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 490, in _execute_search_requests
    raise pre_err
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 481, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=13, reason=worker(13) query failed: [UnexpectedError] Assert "err == simdjson::SUCCESS" at /go/src/github.com/milvus-io/milvus/internal/core/src/common/Json.h:104
 => failed to parse the json : EMPTY: no JSON found)>
```

### Expected Behavior

Search successfully

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility

connections.connect()
dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
json_field = FieldSchema(name="json_field", dtype=DataType.JSON)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, json_field, float_vector])
collection = Collection("test_search_collection_binbin_tmp", schema=schema)
import random
nb=3000
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
import numpy as np
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [{"number": i, "string": "milvus", "bool": True, "list": [j for j in range(0, i)]} for i in range(0, nb)], vectors])
index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}
collection.create_index("float_vector", index_param, index_name="index_name_1")
collection.load()
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
limit = 10
nq =1
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0")
```

**Comment by binbinlv** [comment]
It is good if no list in json field:
```
>>> collection = Collection("test_search_collection_binbin_tmp3", schema=schema)
>>> res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [{"number": i, "string": "milvus", "bool": True} for i in range(0, nb)], vectors])
>>> default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
>>> collection.create_index("float_vector", index_param, index_name="index_name_1")
Status(code=0, message=)
>>> collection.load()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0")
```

**Comment by binbinlv** [comment]
All the expression search report error if there is list in JSON field.

**Comment by yah01** [comment]
The insert node misses to handle JSON data while merging them

**Comment by yah01** [comment]
Tested with #24126 and passed

**Comment by yah01** [comment]
/assign @binbinlv 
fixed with #24126

**Comment by binbinlv** [comment]
Verified and fixed.
milvus: master-20230516-ccd68501
pymilvus: 2.4.0.dev27

```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0", output_fields=["json_field"])
>>>
>>>
>>> res[0][0]
id: 0, distance: 0.0, entity: {}
```

**Comment by binbinlv** [comment]
This issue for 2.2 verification is blocked by issue #24219, will try after #24219 fixed to check whether this exists in 2.2.* branch too.

Reopen this for the tracking on 2.2.*.

**Comment by binbinlv** [comment]
Verified and fixed in 2.2.* branch too:
milvus: [2.2.0-20230520-1eb804fc](https://hub.docker.com/layers/milvusdb/milvus/2.2.0-20230520-1eb804fc/images/sha256-f1fce0b4ba76324119d89cebe860886197da341061e6688f0ef9da34719ef485?context=explore)
pymilvus: 2.2.9.dev12

```
>>> default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
>>> limit = 10
>>> nq =1
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0")
>>> res[0].ids
[0, 145, 1521, 84, 2423, 2671, 179, 1612, 2978, 2856]
```
